### PR TITLE
[IMP] base: new account are lowercased by default

### PR DESCRIPTION
**Pull request by Julien00859**
The account creation and verification was case sensitive on the login
and could cause problems for users to connect (i.e. when a server
administration creates account with random uppercases).

With this commit, newly created accounts will have a lowercase login and
the authentication tests both the user given login (preserving the case)
and lowercase counterpart to ensure backward compatibility with
previously created mixcase accounts plus ease the connection for all new
accounts.

Task: 34047

### CODE REVIEW

File: addons/project/tests/test_access_rights.py [hunk 1]
```diff
@@ -11,8 +11,8 @@ class TestAccessRights(TestProjectCommon):
     def setUp(self):
         super().setUp()
         self.task = self.create_task('Make the world a better place')
-        self.user = new_test_user(self.env, 'Internal user', groups='base.group_user')
-        self.portal = new_test_user(self.env, 'Portal user', groups='base.group_portal')
+        self.user = new_test_user(self.env, 'internal user', groups='base.group_user')
```
Comment (mart-e) [inline addons/project/tests/test_access_rights.py]: other subject but having spaces inside a login doesn't seem a good idea either

**Comment by Julien00859** [comment]
What if the constraint fail ?

**Comment by Julien00859** [comment]
@Julien00859 onchange + test against portal

**Comment by JKE-be** [comment]
You should probably validate it with @odony or @xmo-odoo 

Because I already try a similar patch for trailing whitespace in the past.
And there are more complex situation than just the login via UI. (Ldap, rpc, ...)

In some situation, the case is probably important. And you could have one_login != One_login so you will check in lowercase first, and match a wrong user.

Else for code, js should probably use the events attributes instead to bind manually in the start.

**Comment by fpodoo** [comment]
Not a bad idea, but not a priority (and I believe RFC is case sensitive)

**Pull request closed without merging**